core/src/main/java/org/renjin/primitives/text/Text.java

```diff
@@ -203,6 +203,25 @@ public static void bindtextdomain(String domain, String dirname) {
     // translation not yet supported.
   }
 
+  @Internal
+  public static StringVector enc2utf8(SEXP inputVector) {
+    if (!(inputVector instanceof StringVector)) {
+      throw new EvalException("Error in enc2utf8(" + inputVector + ") : argumemt is not a character vector");
+    }
+
+    StringVector.Builder outputVector = new StringVector.Builder();

```

Comment:
Why are you making a copy here? And why are you handling `length() > 1` and `length() <= 1` cases separately?